### PR TITLE
Implement include/exclude_ids in guide star selection

### PR DESCRIPTION
**Pull request by jeanconn**
Implement include/exclude_ids in guide star selection

### CODE REVIEW

**Comment by jeanconn** [comment]
This was the most direct hack to accomplish the behavior.  I didn't have in mind this use case when doing the original implementation, so it could really use a rewrite to have functions to manipulate the list of candidate stars.

File: proseco/guide.py [hunk 1]
```diff
@@ -261,53 +266,84 @@ def get_initial_guide_candidates(self):
         stars = self.stars
         dark = self.dark
 
-        # Use the primary selection filter from acq, but allow bad color
-        # and limit to brighter stars
-        ok = ((stars['CLASS'] == 0) &
```
Comment (taldcroft) [inline proseco/guide.py]: A corollary to the recommendation of using `ok` for boolean masks is that this should only be done for code that is immediately adjacent.  So move this statement down to [where it is used](https://github.com/sot/proseco/blob/1c14d0e1926aa105aa243b72ec3606d9bb7794ea/proseco/guide.py#L289).

**Comment by jeanconn** [comment]
Thanks @taldcroft .  
-  Forgot to mark as WIP
-  I think you are basically right about implementation
-  My pre-this-hack version was a bit more like what you are describing. But since I annotate the larger set of candidates with information about how they behaved in the checks somewhat as a side effect of running those checks,  my brain was not happy with not having values for those checks for a forced candidate.  Thus, this first full implementation carried along the forced candidates and put them through the checks (to have more complete information for the report).  However, for the special case of forced stars it probably makes more sense to just handle debug and warnings from the starcheck side.

**Comment by jeanconn** [comment]
I've just updated this to be more consistent with @taldcroft 's vision.  There are still compromises in some code that should be addressed in guide star selection, but this is at least a more lean change.

**Comment by jeanconn** [comment]
I ended up not adding back in code to check that include_ids aren't in exclude_ids.

File: proseco/tests/test_catalog.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def test_get_aca_catalog_20603():
     """Put it all together.  Regression test for selected stars.
     """
     # Force not using a bright star so there is a GUI-only (not BOT) star
-    aca = get_aca_catalog(20603, exclude_ids=[40113544], n_fid=2, n_guide=6, n_acq=7,
+    aca = get_aca_catalog(20603, exclude_ids_acq=[40113544], n_fid=2, n_guide=6, n_acq=7,
```
Comment (taldcroft) [inline proseco/tests/test_catalog.py]: Related to the discussion in #178, maybe revert this test and then fix the expected catalog to show both guide and acq exclusion of ..3544?

Comment (taldcroft) [reply proseco/tests/test_catalog.py]: OK fair enough.

**Pull request merged**